Corona-Warn-App/src/main/java/de/rki/coronawarnapp/datadonation/analytics/modules/clientmetadata/ClientVersionParser.kt

```diff
@@ -0,0 +1,37 @@
+package de.rki.coronawarnapp.datadonation.analytics.modules.clientmetadata
+
+import dagger.Reusable
+import de.rki.coronawarnapp.server.protocols.internal.ppdd.PpaData
+import javax.inject.Inject
+
+@Reusable
+class ClientVersionParser @Inject constructor() {
+    fun parseClientVersion(buildNumber: Int): ClientVersion {
+        val major = buildNumber / MAJOR_OFFSET
```

Comment:
Nice

Reply:
If we change the way the version is calculated this needs to change too. Please add a test case that fails if the buildNumber pattern changes, e.g. gets larger for different offsets.

Reply:
makes sense, added a test that compares a generated version string with the original version string 👍